### PR TITLE
Icon: Add support for new semantic colors

### PR DESCRIPTION
**Pull request by ayeshakmaz**
#### What changed?

Add new semantic color options to Icon in preparation for deprecating the old options.

#### Why?

We want to limit icons in Pinterest to semantic, purposeful options that allow for dark mode support.

New colors:
Screen Shot 2022-04-15 at 4 49 24 PM

### Links

- [Jira](https://jira.pinadmin.com/browse/GESTALT-4088)

### Checklist

- [x] Added unit and Flow Tests
- [x] Added documentation + accessibility tests
- [x] Verified accessibility: keyboard & screen reader interaction
- [x] Checked dark mode, responsiveness, and right-to-left support
- [x] Checked stakeholder feedback (e.g. Gestalt designers)

### CODE REVIEW

File: docs/pages/icon.js [hunk 1]
```diff
@@ -136,9 +136,27 @@ If an icon has a visible label that describes what the icon represents, \`access
       </MainSection>
       <MainSection name="Localization" description="Be sure to localize `accessibilityLabel`." />
       <MainSection name="Variants">
-        <MainSection.Subsection title="Primary-color combinations">
-          <CombinationNew color={['gray', 'darkGray', 'red']}>
-            {({ color }) => <Icon icon="heart" accessibilityLabel="" color={color} />}
+        <MainSection.Subsection
```
Comment (ayeshakmaz) [inline docs/pages/icon.js:139]: Main change in docs here

File: packages/gestalt/src/Colors.css [hunk 2]
```diff
@@ -180,6 +180,44 @@
   color: var(--color-text-dark);
 }
 
+/* semantic icon colors */
```
Comment (ayeshakmaz) [inline packages/gestalt/src/Colors.css:183]: New css classes

File: packages/gestalt/src/Datapoint.js [hunk 3]
```diff
@@ -52,8 +52,6 @@ type Props = {|
 /**
  * [Datapoint](https://gestalt.pinterest.systems/datapoint) displays at-a-glance data for a user to quickly view key metrics.
  *
- * ⚠️ Please note: Datapoint is not currently supported in dark mode.
```
Comment (ayeshakmaz) [inline packages/gestalt/src/Datapoint.js:55]: Yay!

File: packages/gestalt/src/Status.js [hunk 4]
```diff
@@ -8,31 +8,31 @@ import Text from './Text.js';
 const ICON_COLOR_MAP = {
   canceled: {
     icon: 'workflow-status-canceled',
-    color: 'gray',
+    color: 'subtle',
```
Comment (ayeshakmaz) [inline packages/gestalt/src/Status.js:11]: utilize the new Icon colors as a proof of concept/test

File: packages/gestalt/src/Icon.js [hunk 5]
```diff
@@ -104,10 +125,23 @@ export default function Icon({
...
 }: Props): Node {
+  let colorClass = null;
+  const colorName = semanticColors.includes(color) ? `${color}Icon` : color;
+  if (
+    colorName !== 'dark' &&
+    colorName !== 'error' &&
+    colorName !== 'light' &&
+    colorName !== 'subtle' &&
+    colorName !== 'success' &&
+    colorName !== 'warning'
```
Comment (AlbertCarreras) [inline packages/gestalt/src/Icon.js:136]: @ayeshakmaz , this seem to be missing. Any reason?

  | 'default'
  | 'inverse'
  | 'shopping'

**Comment by AlbertCarreras** [comment]
I see an issue with color dark

Screen Shot 2022-04-18 at 3 53 24 PM

**Review by AlbertCarreras — APPROVED** [review]
LGTM, 2 points to clarify though

File: packages/gestalt/src/Icon.js (hunk 5 above)
Comment (ayeshakmaz) [reply packages/gestalt/src/Icon.js:136]: It's because they exist in Colors.css, so they don't create the Flow issue

**Comment by ayeshakmaz** [comment]
@AlbertCarreras yeah the combination cmp needs some updates, but it will be easier to do when we remove the old colors, so I'm waiting a bit to refactor that component

**Pull request merged**